sklearn/linear_model/tests/test_logistic.py

```diff
@@ -1136,3 +1136,38 @@ def test_saga_vs_liblinear():
                 liblinear.fit(X, y)
                 # Convergence for alpha=1e-3 is very slow
                 assert_array_almost_equal(saga.coef_, liblinear.coef_, 3)
+
+
+def test_dtype_match():
+    # Test that np.float32 input data is not cast to np.float64 when possible
+
+    X_ = np.array(X).astype(np.float32)
+    y_ = np.array(Y1).astype(np.float32)
+
+    for solver in ['newton-cg']:
+        for multi_class in ['ovr', 'multinomial']:
+
+            # Check type consistency
+            lr_32 = LogisticRegression(solver=solver, multi_class=multi_class)
+            lr_32.fit(X_, y_)
+            assert_equal(lr_32.coef_.dtype, X_.dtype)
+
+            # Check accuracy consistency
+            lr_64 = LogisticRegression(solver=solver, multi_class=multi_class)
+            lr_64.fit(X, Y1)
```

Comment:
Can you ensure (maybe using `astype`?) `X, Y1` are of `float64` before this test? (If in future it is changed, this test will still pass)